lib/money/money.rb

```diff
@@ -548,8 +580,8 @@ def as_euro
   #   Money.new(100, "USD").allocate(3) #=> [Money.new(34), Money.new(33), Money.new(33)]
   #
   def allocate(parts)
-    amounts = Money::Allocation.generate(fractional, parts, !Money.infinite_precision)
-    amounts.map { |amount| self.class.new(amount, currency) }
```

Comment:
Was it an existing bug, or intentional behavior, that the `bank` was not propagated to the new values?

Reply:
I believe it was a bug, I can't any reason why would it be the desired behavior.